src/redux/books/books.js

```diff
@@ -1,42 +1,59 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const FETCHED_BOOK = 'bookstore/books/FETCHED_BOOK';
 const ADDED_BOOK = 'bookstore/books/ADDED_BOOK';
 const REMOVED_BOOK = 'bookstore/books/REMOVED_BOOK';
 
-const initialState = {
-  books: [{
-    id: '1',
-    title: 'The Hunger Games',
-    author: 'Suzanne Collins',
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps';
+const appId = 'coi5KbwSmZBfENXVGL0y';
+const appResourceUrl = `${baseUrl}/${appId}/books`;
+
+const initialState = [];
+
+const fetchBooks = createAsyncThunk(
+  FETCHED_BOOK,
+  async (post, { dispatch }) => {
+    const response = await fetch(appResourceUrl);
+    const jsonData = await response.json();
+    const books = Object.keys(jsonData).map((id) => ({
+      ...jsonData[id][0],
+      item_id: id,
+    }));
+    dispatch({
+      type: FETCHED_BOOK,
+      payload: books,
+    });
   },
-  {
-    id: '2',
-    title: 'Dune',
-    author: 'Frank Herbert',
+);
+
+const addBook = createAsyncThunk(
+  ADDED_BOOK,
+  async (book, { dispatch }) => {
+    await fetch(appResourceUrl, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(book),
+    });
+    dispatch({ type: ADDED_BOOK, payload: book });
   },
-  {
-    id: '3',
-    title: 'The River and The Source',
-    author: 'Thomas Cook',
-  }],
-};
+);
 
-export const addBookAction = (newBook) => ({ type: ADDED_BOOK, newBook });
-export const removeBookAction = (id) => ({ type: REMOVED_BOOK, id });
+const deleteBook = createAsyncThunk(
+  REMOVED_BOOK,
+  async (id, { dispatch }) => {
+    await fetch(`${appResourceUrl}/${id}`, { method: 'DELETE' });
+    dispatch({ type: REMOVED_BOOK, payload: id });
+  },
+);
```

Comment:
- [ ] Wonderful job so far. However, there's an issue with the app at this moment. The expected behaviour of the application is to change the state of the app once an action is performed, not by performing double actions. What I am saying is this. When delete button is clicked the expected behaviour is for the state to change from the redux store. But the actual action happens when the page  is refreshed. Kindly look into it and fix.

[page-behaviour.webm](https://user-images.githubusercontent.com/26597432/216650146-b0cd9498-afa4-4d4b-b4c0-6c8537edec8c.webm)

Reply:
@Qoosim Thank you so much.